common/beerocks/bwl/dwpal/base_wlan_hal_dwpal.cpp

```diff
@@ -628,7 +628,7 @@ size_t base_wlan_hal_dwpal::dwpal_nl_cmd_get(const std::string &ifname, unsigned
                                              unsigned char *out_buffer,
                                              const size_t max_buffer_size)
 {
-    size_t data_size = 0;
+    size_t data_size = -1;
 
     if (!out_buffer) {
         LOG(ERROR) << "out_buffer is invalid ==> Abort!";
```

Comment:
`data_size` is -1 and never changed until the following while loop is used:
`while ((data_size == 0) && std::chrono::steady_clock::now() < timeout) {`
this condition will never be met.

did you test this flow?

anyways I prefer to return -1 and not data_size in case of failure - instead of changing data_size to -1 just for that

1. return `-1` instead of the `data_size on` failure
2. set data_size back to `0` and you can move the point of declaration to above the lambda (close to where it will be first used after the above-mentioned change)

Reply:
Currently the function returns `ssize_t` which accepts either a positive `size_t` value or `-1`
And for each of the error handling in the function I returned `-1` to signal an error